samcli/lib/telemetry/metric.py

```diff
@@ -170,6 +177,62 @@ def wrapped(*args, **kwargs):
     return wrapped
 
 
+def _get_stack_trace_info(exception: Exception) -> Tuple[str, str]:
+    """
+    Takes an Exception instance and extracts the following:
+      1. Stack trace in a readable string format with user-sensitive paths cleaned
+      2. Exception mesage including the fully-qualified exception name and value
+
+    Parameters
+    ----------
+    exception : Exception
+        Exception instance
+
+    Returns
+    -------
+    (str, str)
+        (stack trace, exception message)
+    """
+    tb_exception = traceback.TracebackException.from_exception(exception)
+    _clean_stack_summary_paths(tb_exception.stack)
+    stack_trace = "".join(list(tb_exception.format()))
+    exception_msg = list(tb_exception.format_exception_only())[-1]
```

Comment:
Are we sure the list always has at least 1 element?

Reply:
Yup: https://docs.python.org/3/library/traceback.html#traceback.format_exception_only